lib/runtime-core/src/vmcalls.rs

```diff
@@ -11,89 +13,106 @@ use crate::{
 
 pub unsafe extern "C" fn local_static_memory_grow(
     memory_index: LocalMemoryIndex,
-    by_pages: u32,
-    ctx: *mut vm::Ctx,
+    delta: u32,
+    ctx: &mut vm::Ctx,
 ) -> i32 {
-    if let Some(old) = (*(*ctx).local_backing)
-        .memory(memory_index)
-        .grow_static(by_pages)
-    {
-        // Store the new size back into the vmctx.
-        (*(*ctx).memories.add(memory_index.index())).size =
-            (old as usize + by_pages as usize) * LinearMemory::PAGE_SIZE as usize;
-        old
+    let local_memory = *ctx.memories.add(memory_index.index());
+    let memory = (*local_memory).memory as *mut StaticMemory;
+
+    if let Some(old) = (*memory).grow(delta, &mut *local_memory) {
+        old as i32
     } else {
         -1
     }
 }
 
 pub unsafe extern "C" fn local_static_memory_size(
     memory_index: LocalMemoryIndex,
-    ctx: *mut vm::Ctx,
+    ctx: &vm::Ctx,
 ) -> u32 {
-    (*(*ctx).local_backing).memory(memory_index).pages()
+    let local_memory = *ctx.memories.add(memory_index.index());
+    let memory = (*local_memory).memory as *mut StaticMemory;
+
+    (*memory).current()
 }
 
 pub unsafe extern "C" fn local_dynamic_memory_grow(
     memory_index: LocalMemoryIndex,
-    by_pages: u32,
-    ctx: *mut vm::Ctx,
+    delta: u32,
+    ctx: &mut vm::Ctx,
 ) -> i32 {
-    if let Some(old) = (*(*ctx).local_backing)
-        .memory(memory_index)
-        .grow_dynamic(by_pages)
-    {
-        // Store the new size back into the vmctx.
-        (*(*ctx).memories.add(memory_index.index())).size =
-            (old as usize + by_pages as usize) * LinearMemory::PAGE_SIZE as usize;
-        old
+    let local_memory = *ctx.memories.add(memory_index.index());
+    let memory = (*local_memory).memory as *mut DynamicMemory;
+
+    if let Some(old) = (*memory).grow(delta, &mut *local_memory) {
```

Comment:
Is there a way to make this less verbose? The js API for this is `memory.grow(1)`

Reply:
The user can just call `memory.grow(delta)`. This cruft in this function is because it's specifically accessing a particular type of memory. The user-exposed api doesn't expose that.